### PR TITLE
Remove support for bypassing linking errors.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/Refiner.scala [hunk 1]
```diff
@@ -31,38 +31,30 @@ final class Refiner {
...
+    /* There must not be linking errors at this point. If there are, it is a
+     * bug in the optimizer.
      */
-    analysis.errors.foreach(Analysis.logError(_, logger, Level.Warn))
+    if (analysis.errors.nonEmpty) {
+      analysis.errors.foreach(Analysis.logError(_, logger, Level.Error))
+      throw new AssertionError(
+          "There were linking errors after the optimizer has run. " +
+          "This is a bug, please report it. " +
+          "You can work around the bug by disabling the optimizer.")
```
Comment (gzm0) [inline tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/Refiner.scala]: Maybe say how?

Comment (sjrd) [inline tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/Refiner.scala]: Done.

**Comment by sjrd** [comment]
Updated with the "say how", and also removal of the option in `Scalajsld.scala`.

**Comment by sjrd** [comment]
Rebased to satisfy Jenkins.

**Comment by sjrd** [comment]
Updated with an additional commit that blacklists two partest using .java files.

**Pull request merged**